codeforamerica/il-gcc-form-flow
# CCAP 374 Create transmission table

#### 🔗 Jira ticket
https://codeforamerica.atlassian.net/browse/CCAP-374?atlOrigin=eyJpIjoiOTQwYzg0ZGMxY2Y3NGUxNzhmZWE1MTYyYzJkODhiYTYiLCJwIjoiaiJ9

#### ✍️ Description
Adds a transmissions table to track transmission of Application PDFs and Uploaded Documents for a given Submission.

Adds enums for TransmissionStatus and TransmissionType which will be used later when saving new transmissions in a separate ticket. 

####
To test the migration:

0. Check out main
1. Remove all user files from your local user_files table
2. Remove all submissions from your local submissions table
3. With a clean user_files and submissions table - run through the application a few times. Once with 2 or 3 uploaded documents, once with no uploaded documents, and once without reaching the submission screen.
4. Check out this branch and restart the app so the migration runs
5. When the migration runs, refresh your local DB setup and you should now see the transmissions table
6. You should see one transmission for each completed submissions application PDF and one transmission for each uploaded document related to a submission.
7. You should not see any transmission for submissions that were not complete (have a null submitted at)
